ext/rugged/rugged_object.c

```diff
@@ -132,11 +134,56 @@ git_object *rugged_object_get(git_repository *repo, VALUE object_value, git_otyp
 	return object;
 }
 
-static void rb_git_object__free(git_object *object)
+static void rb_git_object__free(void *data)
 {
+	git_object *object = (git_object *) data;
 	git_object_free(object);
 }
 
+static size_t rb_git_object__size(const void *data)
+{
+	git_object *object = (git_object *) data;
+	size_t size = 0;
+
+	/*
+	 * We cannot always be accurate cheaply, but we can give some numbers
+	 * which are closer than zero by taking an average/guessed size.
+	 */
+	switch (git_object_type(object)) {
+	case GIT_OBJ_BLOB:
+	{
+		git_blob *blob = (git_blob *) object;
+		size = git_blob_rawsize(blob);
+		break;
+	}
+	case GIT_OBJ_TREE:
+	{
+		git_tree *tree = (git_tree *) object;
+		size = git_tree_entrycount(tree) * 64;
+		break;
+	}
+	case GIT_OBJ_COMMIT:
+	case GIT_OBJ_TAG:
+		size = 256;
+		break;
+	default:
+		break;
+	}
+
+	return size;
+}
+
+const rb_data_type_t rugged_object_type = {
+	.wrap_struct_name = "Object",
```

Comment:
This is a global identifier and doesn't need to be valid Ruby or C, so calling this `Rugged::Object` would be good: otherwise we're gonna see a lot of random "object" in our heap dumps.

Reply:
Yep, that sounds good.